src/Repository/AbstractDatabaseMessageRepository.php

```diff
@@ -206,7 +212,11 @@ public function publishOrUpdateEntityMessage(Message\MessageInterface $message):
         $existingMessage = $this->getMessage($filterExisting);
 
         if ($existingMessage instanceof Message\MessageInterface) {
-            $this->mergeMessages($existingMessage, $message);
+            if ($mergeCallback !== null) {
+                $mergeCallback($existingMessage, $message);
```

Comment:
wouldn't it make more sense to have the regular merge first, then apply the callback ?
developers wouldn't have to worry about the basic merge behavior, but just about what they want to happen with their payload

Reply:
I hesitated to do that.

But yes, you're right, it is probably better to pre-apply base merging.